### PR TITLE
Configure logging from environment using viper

### PR DESCRIPTION
**Pull request by MarkGreenPR**
Need to agree on and finalize default values #3

### CODE REVIEW

File: go/logger/config.go [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+	}
+	kafkaConfig := new(ProducerConfiguration)
+	err = EnvConfigure(defaultKafkaCfg(), kafkaConfig, os.Getenv(KafkaAutoCfgEnvName),
+		KafkaFileName, KafkaEnvPrefix)
+	// fmt.Printf("config %+v\n\n", config)
+	if err != nil {
+		fmt.Printf("Could not create kafka configuration %s:", err.Error())
+		os.Exit(1)
+	}
+	config.KafkaProducerCfg = *kafkaConfig
```
Comment (jscharber) [inline go/logger/config.go]: Likely we will make Kafka logging optional.  I don't see a way to do that.  Even if it is configured, do we want to exit if the config can't be read?

File: go/logger/config.go [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+
+func EnvConfigure(defaultCfg interface{}, config interface{}, auto string,
+	filename string, prefix string) error {
+
+	var defaultMap map[string]interface{}
+	defaultJson, err := json.Marshal(defaultCfg)
+	if err != nil {
+		return err
+	}
+	json.Unmarshal(defaultJson, &defaultMap)
```
Comment (jscharber) [inline go/logger/config.go]: returns an error code that should be checked.

File: go/logger/zap.go [hunk 3]
```diff
@@ -41,17 +41,17 @@ func getEncoder(format FormatType) zapcore.Encoder {
...
+	case ErrorType:
 		return zapcore.ErrorLevel
-	case Fatal:
+	case FatalType:
 		return zapcore.FatalLevel
-	case Panic:
+	case PanicType:
 		return zapcore.PanicLevel
 	default:
 		return zapcore.InfoLevel
```
Comment (jscharber) [inline go/logger/zap.go:66]: Still needs README and test cases.

File: go/logger/config.go (hunk 1 above)
Comment (MarkGreenPR) [reply go/logger/config.go]: Not exiting if Kafka is not enables

File: go/logger/config.go (hunk 2 above)
Comment (MarkGreenPR) [reply go/logger/config.go]: Added error check

File: go/logger/zap.go (hunk 3 above)
Comment (MarkGreenPR) [reply go/logger/zap.go:66]: TBD next branch

**Pull request merged**